src/main/java/org/example/Util/HttpRequestUtil.java

```diff
@@ -15,97 +13,76 @@ public class HttpRequestUtil {
-                .build();
-    }
-
-    public static HttpRequest buildRequestGET(String baseUrl, String token, String form) {
-        return HttpRequest.newBuilder()
-                .uri(URI.create(baseUrl + "?" + form))
-                .header("Authorization", "Bearer " + token)
-                .GET()
-                .build();
-    }
-
-    public static HttpResponse<String> sendRequest(HttpClient client, HttpRequest request)
-            throws IOException, InterruptedException {
-        response = client.send(request, HttpResponse.BodyHandlers.ofString());
-        return response;
-    }
-
-    public static HttpResponse<String> getResponse() {
-        return response;
-    }
-
-    public static <T> T parseResponse(HttpResponse<String> response, Class<T> responseType) throws JsonProcessingException {
-        return mapper.readValue(response.body(), responseType);
-    }
-
     /**
      * Sends a POST HTTP request and parses the JSON using the methods above
-     * @param formData
+     * @param formData Contains the parameters of the HTTP request
      * @param baseUrl
      * @param token
-     * @param responseType Class type of the response object
      * @return Instance of the response type containing the parsed data
-     * @param <T> Type of response object
      * @throws IOException
      * @throws InterruptedException
      */
-    public static <T> T sendAndParsePOST(Map<Object, Object> formData, String baseUrl, String token,
-                                         HttpClient client, Class<T> responseType) throws IOException, InterruptedException {
-        // creating the request body
-        String form = HttpRequestUtil.createRequestBody(formData);
+    public static HttpResponse<String> sendAndParsePOST(Map<Object, Object> formData, String baseUrl, String token,
+                                                        HttpClient client) {
+        /// creating the request body
+        String form = formData.entrySet()
+                .stream()
+                .map(Map.Entry::toString) // method reference to Map.Entry.toString(): key=value
+                .collect(Collectors.joining("&"));
 
         // creating the request
-        HttpRequest request = HttpRequestUtil.buildRequestPOST(baseUrl, token, form);
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(baseUrl))
+                .header("Authorization", "Bearer " + token)
+                .header("Content-Type", "application/x-www-form-urlencoded")
+                .POST(HttpRequest.BodyPublishers.ofString(form))
+                .build();
 
         // sending request
-        HttpResponse<String> response = HttpRequestUtil.sendRequest(client, request);
+        HttpResponse<String> response;
+        try {
+            response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Why catch the exception and wrap in a different exception type?

Reply:
Truthfully I just hit the "surround with try/catch" on my ide when I had an unhandled exception. Would it be better to throw the respective exception?